pkg/executor/wasm/loader.go

```diff
@@ -60,46 +55,38 @@ func (loader *ModuleLoader) Load(ctx context.Context, path string) (wazero.Compi
 }
 
 // LoadRemoteModules loads and compiles all of the modules located by the passed storage specs.
-func (loader *ModuleLoader) LoadRemoteModules(ctx context.Context, specs ...model.StorageSpec) ([]wazero.CompiledModule, error) {
+func (loader *ModuleLoader) LoadRemoteModules(ctx context.Context, m storage.PreparedStorage) (wazero.CompiledModule, error) {
 	ctx, span := system.NewSpan(ctx, system.GetTracer(), "pkg/executor/wasm.ModuleLoader.LoadRemoteModules")
 	defer span.End()
 
-	volumes, err := storage.ParallelPrepareStorage(ctx, loader.provider, specs)
+	programPath := m.Volume.Source
+
+	info, err := os.Stat(programPath)
 	if err != nil {
 		return nil, err
 	}
 
-	var loadedModules []wazero.CompiledModule
-	for _, volume := range maps.Values(volumes) {
-		programPath := volume.Source
-
-		var info fs.FileInfo
-		info, err = os.Stat(programPath)
+	// We expect the input to be a single WASM file. It is common however for
+	// IPFS implementations to wrap files into a directory. So we make a special
+	// case here – if the input is a single file in a directory, we will assume
+	// this is the program file and load it.
+	if info.IsDir() {
+		files, err := os.ReadDir(programPath)
```

Comment:
We should probably correct the method name and method command if we're only returning a single module (previously this method returned several).

Reply:
Good catch, will also make it private as it's only used internally.